internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -592,7 +592,6 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			"private_cluster_public_fqdn_enabled": {
 				Type:     pluginsdk.TypeBool,
 				Optional: true,
-				Default:  false,
```

Comment:
We should probably just make this computed otherwise it could result in cluster rebuilds for some users?

Reply:
The default value is not causing the cluster to be re-created. I put a comment in the issue. I think the default must be kept to false and the forceNew set to false. This is forceNew causing the cluster to be re-created.
This happens as Microsoft are releasing a new feature on private clusters that give an option to expose the private ip of the cluster to a public fqdn. This feature is exposed on the REST API used by version 2.73+ hence causing the cluster to be re-created.
New clusters created from 2.73.0 are created with private_cluster_public_fqdn_enabled = false as per expected behavior with public aks documentation.

Reply:
Removing this default will will set this to the default value for a bool - which is false - so this PR won't have any affect anyway, unfortunately

Is it now possible to change this field without destroying/re-creating the cluster via the API? This was historically the case (and is why this is ForceNew) - so we'd need to confirm that to be able to determine if we can remove ForceNew here